datalake/src/main/java/com/sequenceiq/datalake/logger/MDCContextFilter.java

```diff
@@ -47,11 +49,12 @@ public MDCContextFilter(ThreadBasedUserCrnProvider threadBasedUserCrnProvider) {
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws IOException, ServletException {
         MDCBuilder.cleanupMdc();
         HttpServletRequestWrapper wrapper = new RequestIdHeaderInjectingHttpRequestWrapper(request);
-        MdcContext.builder()
-                .userCrn(threadBasedUserCrnProvider.getUserCrn())
-                .tenant(threadBasedUserCrnProvider.getAccountId())
-                .requestId(wrapper.getHeader(REQUEST_ID_HEADER))
-                .buildMdc();
+        doIfNotNull(threadBasedUserCrnProvider.getUserCrn(), crn ->
+                MdcContext.builder()
+                        .userCrn(crn)
+                        .tenant(threadBasedUserCrnProvider.getAccountId())
+                        .requestId(wrapper.getHeader(REQUEST_ID_HEADER))
```

Comment:
Do you want to skip setting `requestId`, too, if `userCrn` is null?  If so, then shouldn't the log message below (`"Request id has been added ..."`) be skipped, too?

If not, then only `userCrn(crn)` call should be conditional.

Reply:
good point, I fix it